packages/ra-no-code/src/builders/useGetFieldDefinitions.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import {
+    email,
+    getValuesFromRecords,
+    InferredElementDescription,
+    inferTypeFromValues,
+    Record,
+} from 'ra-core';
+import {
+    BooleanField,
+    BooleanInput,
+    DateField,
+    DateInput,
+    EmailField,
+    ImageField,
+    ImageInput,
+    NumberField,
+    NumberInput,
+    TextField,
+    TextInput,
+    UrlField,
+} from 'ra-ui-materialui';
+import inflection from 'inflection';
+
+export const useGetFieldDefinitions = (resource: string, records?: any[]) => {
```

Comment:
why do you need two separate keys (and hooks) to store resource name and fields? Seems a bit over engineered.